samples/matter/lock/README.rst

```diff
@@ -66,6 +69,26 @@ For details, see the `Commissioning the device`_ section.
 
 .. matter_door_lock_sample_remote_testing_end
 
+.. _matter_lock_sample_wifi_thread_switching:
+
+Thread and Wi-Fi switching
+==========================
+
+When built using the ``thread_wifi_switched`` build type and programmed to the nRF5340 DK with the nRF7002 EK shield attached, the door lock sample supports a feature that allows you to dynamically switch between Matter over Thread and Matter over Wi-Fi.
+
+.. matter_door_lock_sample_thread_wifi_switch_desc_start
+
+The application is built in two variants: the first one is the application working with Matter over Thread and the second one is the application working with Matter over Wi-Fi.
+Both application variants are programmed into separate partitions of the external flash.
+The application runs from the internal flash memory, using one of the variants from the external flash.
+You can trigger the switch from one variant to another using the **Button 3** on the nRF5340 DK.
+The device is rebooted into the MCUboot bootloader, which replaces the current variant by swapping the application variant in the internal flash.
+
```

Comment:
Could we add a note, that because external flash is used for both DFU and switching purpose, memory size requirements are different. Specifically it is required to use external flash with at least 6MB of memory.